### ISSUE TITLE
[Feature Request] Allow reserved words as prop names

### ISSUE THREAD

**Issue by alexcjohnson** [plotly/dash#878]
Currently if someone makes a component that uses a python keyword or an R reserved word as a prop name, it just gets dropped from the generated component for that language. Here's the Python part (the important part being `p not in python_keywords`):

https://github.com/plotly/dash/blob/076f5ec92b9bd034d3904e97bd58fc6207482886/dash/development/_py_components_generation.py#L90-L99

Silently dropping some props seems like it's definitely the wrong approach. We *could* fail to generate the component - that would be simple, and I'm not aware of any cases this would break now - but it's a silent failure, so `¯\_(ツ)_/¯` - also we'd have to fail on R keywords, and if we add another language potentially we'd add new keywords that could break existing components.

So I'd propose we find a way to allow keywords to be used as props, likely by altering the prop name / usage on the server side, like by adding a leading underscore? (may need to be language-dependent - leading underscore wouldn't work in R I don't believe) Note this would need to work consistently in both component construction and callback construction.

This came up during discussion of #874

**Comment by alexcjohnson** [comment]
@rpkyle points out that this comes up in `html.Output`, which has a `for` prop in the HTML spec. I can't say that tag looks particularly useful, `html.Output(value=v)` appears to be identical to `html.Span(children=v)`, and `for` in particular just seems to add some simple semantics into the DOM but I don't see any visual effect.

**Comment by jjaraalm** [comment]
This is problematic at least in python. You cannot use reserved words as arguments or keyword arguments, so instead of `Output(for='a')` the only way to do it is `Output(**{'for': 'a'})` which is just ugly and hackish.

For this specific case (html components) the correct and existing syntax is the same as in react, `Output(htmlFor='a')`. `htmlFor` is pretty critical for generating accessible forms.

**Comment by alexcjohnson** [comment]
Right, I don’t think we want to just use it verbatim, ideally I’d like to just modify the names for example with a leading underscore.

**Comment by alexcjohnson** [comment]
Ah sorry, I missed the point that React is already doing this translation in the case of `for`, since this is also a *JavaScript* reserved word - so you're right, in that case we should simply consider it a bug in the `dash_html_components` package and change the prop name to `htmlFor`.

But then that's separate from the consideration of allowing *Python* reserved words (and R, and whatever other languages we add in the future) as prop names, which I still think we should do with a leading underscore for Python.

**Comment by jjaraalm** [comment]
Yes, I was mostly replying to the html case. The `htmlFor` keyword already
exists on the dash components (at least on Label).
